src/main/java/org/wso2/carbon/esb/connector/MQPublishTopic.java

```diff
@@ -0,0 +1,134 @@
+/*
+*  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.carbon.esb.connector;
+
+import com.ibm.mq.MQException;
+import com.ibm.mq.MQMessage;
+import com.ibm.mq.MQQueueManager;
+import com.ibm.mq.MQTopic;
+import com.ibm.mq.constants.CMQC;
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.soap.SOAPEnvelope;
+import org.apache.synapse.MessageContext;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+
+import java.util.Date;
+import java.util.GregorianCalendar;
+
+import static com.ibm.mq.constants.CMQC.MQPER_NOT_PERSISTENT;
+import static com.ibm.mq.constants.CMQC.MQPER_PERSISTENT;
+
+/**
+ * Add messages to queue
+ */
+public class MQPublishTopic extends AbstractConnector {
+
+    MQConnectionBuilder connectionBuilder;
```

Comment:
This will not work for the concurrent cases. Please see related comments for queue, same goes here as well.